enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -0,0 +1,1104 @@
+Once this is copied, they should remove the failure domain and add the desired failure domains to the `FailureDomains`
+field within the ControlPlaneMachineSet spec.
+
+To ensure adding a `ControlPlaneMachineSet` to the cluster is safe, we will need to ensure via a validating webhook
+that the replica count defined in the spec is consistent with the actual size of the control plane within the cluster.
+We will also validate that the failure domains align with those within the cluster already, this will prevent users
+from accidentally migrating their Control Plane Machines from multiple availability zones to a single zone.
+
+If no Control Plane Machines exist, or they are in a non-Running state, the operator will report degraded until this
+issue is resolved. This creates a dependency for the `ControlPlaneMachineSet` operator on the Machine API. It will be
+required to run at a higher run-level than Machine API.
+
+This in turn means, that in a UPI (where typically Machine objects do not exist) or misconfigured clusters, a user
+adding a `ControlPlaneMachineSet` will result in a degraded cluster.
+Users will need to remove the invalid `ControlPlaneMachineSet` resource, or manually add correctly configured Control
+Plane Machines to their cluster to restore their cluster to a healthy state.
+
+We do not recommend that UPI users attempt to adopt their control plane instances into Machine API due to the
+likelihood that the resulting spec does not match the existing Machines. Instead, we recommend that UPI users initially
+go through the process of replacing their control plane instances with Machines by creating new Control Plane Machines,
+allowing Machine API to create the instances, and then removing their old, manually created control plane instance.
+This effectively means migrating their entire control plane onto Machine API before the `ControlPlaneMachineSet` will
+take over the management.
+We enforce this recommendation so that users can be confident in the Machine `providerSpec` that they have configured
+and that Machine API will be able to create valid Machines from the spec before they then start using our automation.
+
+##### Why not to populate the spec for the customer
+
+One idea posed, is to allow the `ControlPlaneMachineSet` to be automatically populated if the spec is left blank when it
+is created. This would improve the UX by allowing customers not to have to worry about the spec and allow it to be
+inferred from the existing Machines within the cluster. It would also make it easier for managed services to adopt
+`ControlPlaneMachineSets` throughout the fleet without having to manually check each cluster.
+
+There are however a few concerns about this idea which means we are planning to not implement this (at least for the
+first iteration of the project):
+- It may encourage users to attempt to use `ControlPlaneMachineSets` with UPI clusters which are incorrectly configured
```

Comment:
The controller is designed in a way such that, without an existing working set of Control Plane Machines it will degrade and not take any action, we will inform the user via the status of actions they need to take to resolve the issue.

So yes, unless the user has correctly configured a set of Machines, we will not do anything, in the same way the etcd protection does.